src/components/metadataSelects/CategoryComboSelect.tsx

```diff
@@ -0,0 +1,132 @@
+            }
+
+            if (dynamicParams.filter) {
+                baseParams.filter = `displayName:ilike:${dynamicParams.filter}`
+            }
+
+            return baseParams
+        },
+    },
+}
+
+interface Option { value: string, label: string }
+export function CategoryComboSelect({
+    onChange,
+    selected,
+}: {
+    onChange: ({ value, label }: Option) => void,
+    // Must be an option as we always must supply the selected option, but the
+    // list might not contain it when filtering / loading the first page of
+    // options
+    selected?: Option,
+}) {
+    // Using a ref because we don't want to react to a change of this value
+    // It's guaranteed that we have the correct value because we reset the
+    // value before fetching a new set of options when filtering / resetting
+    // the filter as well as setting the value after the fetch has completed
+    // but before we call a state setter. The state setter will cause a
+    // rerender during which the new pages value can be accessed already
+    const pages = useRef(1)
+    const [loadedOptions, setLoadedOptions] = useState<Array<{
+      value: string,
+      label: string,
+    }>>([])
+
+    const [params, setParams] = useState({ page: 1, filter: '' })
+    const filterTimeout = useRef<ReturnType<typeof setTimeout>>()
+    const adjustQueryParamsWithChangedFilter = useCallback(({ value }: { value: string }) => {
+        clearTimeout(filterTimeout.current)
+        filterTimeout.current = setTimeout(() => {
+            pages.current = 1
+            setLoadedOptions([])
+            setParams({ page: 1, filter: value })
+        }, 200)
+    }, [])
+
+    const incrementPage = ({ isIntersecting }: { isIntersecting: boolean }) => {
+        if (!isIntersecting) {
+            return false
+        }
+
+        setParams((prevParams) => {
+            const prevPage = prevParams.page
+            const nextPage =
+                prevPage < pages.current ? prevPage + 1 : prevPage
+            return { ...prevParams, page: nextPage }
+        })
+    }
+
+    const queryResult = useDataQuery<CategoryComboQueryResult>(query, {
+        lazy: true,
+        onComplete: (data) => {
+            pages.current = data.categoryCombos.pager.pageCount
+            // We want to add new options to existing ones so we don't have to
+            // refetch existing options
+            setLoadedOptions((prevLoadedOptions) => [
+                ...prevLoadedOptions,
+                ...data?.categoryCombos.categoryCombos.map(({ id, displayName }) => ({
+                    value: id,
+                    label: displayName,
+                })) || [],
+            ])
+        },
+    })
+
+    const { refetch } = queryResult
+    useEffect(() => {
+        refetch(params)
+    }, [params, refetch])
+
+    const actualOptions = !selected || loadedOptions.find(({ value }) => value === selected?.value)
+        ? loadedOptions
+        : [selected, ...loadedOptions]
+
```

Comment:
Currently the list will flash when searching. We should keep the previous state of the list while a fetch is pending. 
This will be easier if/when (hopefully) we use `react-query` for fetching with `keepPrevious`-data.

Reply:
I was thinking about a different solution:

Instead of adding the selected value to the list (which makes it more difficult to render the dropdown, e.g. because we have to fetch the label when we have the ID only), we change the `selected` prop to: ```{ selected: string }```

Then we can render an invisible option, I guess it's fine to not render the selected option when searching the options as the selected value can be seen anyway

Reply:
Hm... I don't really see how that fixes the flashing, though. That is related to paging and combining  the result, shown in the comment below. Think my comment was placed wrong here.

Reply:
This is not needed anymore. Right now we don't display the selected option when it's not in the list